src/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.cs

```diff
@@ -12,6 +12,8 @@ namespace Microsoft.Extensions.DependencyModel
 {
     public class DependencyContextJsonReader : IDependencyContextReader
     {
+        private readonly IDictionary<string, string> _stringPool = new Dictionary<string, string>();
```

Comment:
Ya, good point.  I didn't fully think it through. You can check if the string is already cached, but there isn't a good way of getting the cached string out.

It just seemed odd that we have a Dictionary where the key == value. And that there isn't a more efficient data structure for this. You don't really need to track both the key and value. You just need to track the key...